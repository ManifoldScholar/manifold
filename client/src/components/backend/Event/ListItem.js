import React, { PureComponent } from "react";
import PropTypes from "prop-types";
import { Event } from "components/frontend";

export default class EventListItem extends PureComponent {
  static displayName = "Event.ListItem";

  static propTypes = {
    entity: PropTypes.object,
    destroyHandler: PropTypes.func
  };

  constructor(props) {
    super(props);
    this.triggerDestroy = this.triggerDestroy.bind(this);
  }

  triggerDestroy(event) {
    event.preventDefault();
    this.props.destroyHandler(this.props.entity);
  }

  render() {
    const event = this.props.entity;
    if (!event) return null;
    return (
      <Event.Event
        itemClass="list-item"
        event={event}
        hideLink
        destroyCallback={this.triggerDestroy}
      />
    );
  }
}
