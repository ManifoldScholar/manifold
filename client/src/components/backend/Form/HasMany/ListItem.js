import React, { PureComponent } from "react";
import PropTypes from "prop-types";
import get from "lodash/get";

export default class FormHasManyList extends PureComponent {
  static displayName = "Form.HasMany.List";

  static propTypes = {
    entity: PropTypes.object.isRequired,
    entities: PropTypes.array,
    entityAvatarAttribute: PropTypes.string,
    ordinal: PropTypes.number,
    entityName: PropTypes.func,
    label: PropTypes.string,
    orderable: PropTypes.bool,
    removeHandler: PropTypes.func.isRequired,
    moveHandler: PropTypes.func.isRequired,
    editHandler: PropTypes.func
  };

  onRemove = event => {
    return this.props.removeHandler(event, this.props.entity);
  };

  onEdit = event => {
    event.preventDefault();
    return this.props.editHandler(this.props.entity);
  };

  onMoveUp = event => {
    return this.props.moveHandler(event, this.props.entity, "up");
  };

  onMoveDown = event => {
    return this.props.moveHandler(event, this.props.entity, "down");
  };

  name() {
    return this.props.entityName(this.props.entity);
  }

  renderAvatar() {
    if (!this.props.entityAvatarAttribute) return null;
    const path = `${this.props.entityAvatarAttribute}.smallSquare`;
    const avatar = get(this.props.entity.attributes, path);

    return (
      <figure>
        {avatar ? (
          <img src={avatar} alt="User Avatar" />
        ) : (
          <div className="no-image">
            <i className="manicon manicon-person" aria-hidden="true" />
          </div>
        )}
      </figure>
    );
  }

  renderEditButton() {
    if (!this.props.editHandler) return null;

    return (
      <button className="manicon manicon-pencil-simple" onClick={this.onEdit}>
        <span className="screen-reader-text">
          Edit {this.name()}
          in the {this.props.label} list.
        </span>
      </button>
    );
  }

  renderRemoveButton() {
    return (
      <button onClick={this.onRemove} className="manicon manicon-x">
        <span className="screen-reader-text">
          Remove {this.name()} from the {this.props.label} list.
        </span>
      </button>
    );
  }

  renderOrderButton(direction, ordinal) {
    if (!this.props.orderable) return null;
    if (direction === "up" && ordinal === 0) return null;
    if (direction === "down" && ordinal === this.props.entities.length - 1)
      return null;

    const moveHandler = direction === "up" ? this.onMoveUp : this.onMoveDown;

    return (
      <button
        onClick={moveHandler}
        className={`manicon manicon-arrow-${direction}`}
      >
        <span className="screen-reader-text">
          {`Move ${this.name()} ${direction} in the ${this.props.label} list`}
        </span>
      </button>
    );
  }

  render() {
    const { entity, ordinal } = this.props;
    if (!entity) return null;

    return (
      <li>
        <div className="association">
          {this.renderAvatar()}
          <h4 className="association-name">{this.name()}</h4>
        </div>

        <div className="utility">
          {this.renderEditButton()}
          {this.renderOrderButton("up", ordinal)}
          {this.renderOrderButton("down", ordinal)}
          {this.renderRemoveButton()}
        </div>
      </li>
    );
  }
}
