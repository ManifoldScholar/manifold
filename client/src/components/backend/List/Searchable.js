import React, { PureComponent, PropTypes } from 'react';
import { Utility, Project as globalProject } from 'components/global';
import { Link } from 'react-router';
import get from 'lodash/get';

export default class ListSearchable extends PureComponent {

  static displayName = "List.Searchable";

  static propTypes = {
    entities: PropTypes.array,
    singularLabel: PropTypes.string,
    pluralLabel: PropTypes.string,
    pagination: PropTypes.object,
    paginationClickHandler: PropTypes.func
  };

  static defaultProps = {
    entityComponentProps: {}
  }

  constructor() {
    super();

    this.state = this.initialState();
    this.setKeyword = this.setKeyword.bind(this);
    this.resetSearch = this.resetSearch.bind(this);
    this.renderEntityList = this.renderEntityList.bind(this);
    this.renderEntity = this.renderEntity.bind(this);
  }

  initialState() {
    return {
      inputs: { keyword: "" },
      filter: { }
    };
  }

  componentDidUpdate(prevProps, prevState) {
    if (get(prevState, 'filter.keyword') !== get(this.state, 'filter.keyword')) {
      this.props.filterChangeHandler(this.state.filter);
    }
  }

  resetSearch(event) {
    event.preventDefault();
    this.setState(this.initialState());
  }

  setKeyword(event) {
    const keyword = event.target.value;
    const filter = Object.assign({}, this.state.filter);
    if (keyword === "") {
      delete filter.keyword;
      delete filter.typeahead;
    } else {
      filter.keyword = keyword;
      filter.typeahead = true;
    }
    this.setState({ inputs: { keyword }, filter });
  }

  renderEntity(entity) {
    const props = Object.assign({}, this.props.entityComponentProps);
    props.key = entity.id;
    props.entity = entity;
    return React.createElement(this.props.entityComponent, props);
  }

  renderEntityList() {
    const entities = this.props.entities;
    if (!entities) return;

    let output = null;

    if (entities.length > 0) {
      output = (
        <div>
          <Utility.EntityCount
            pagination={this.props.pagination}
            singularUnit={this.props.singularUnit}
            pluralUnit={this.props.pluralUnit}
          />
          <ul>
            {entities.map((entity) => {
              return this.renderEntity(entity);
            })}
          </ul>
        </div>
      );
    } else {
      output = (
        <p className="list-total">Sorry, no results were found.</p>
      );
    }

    return output;
  }

  render() {
    return (
      <div>
        <form className="form-search-filter">
          <div className="search">
            <button>
              <i className="manicon manicon-magnify"></i>
              <span className="screen-reader-text">Click to search</span>
            </button>
            <input
              value={this.state.inputs.keyword}
              type="text"
              placeholder="Search..."
              onChange={this.setKeyword}
            />
          </div>
          <button
            onClick={this.resetSearch}
            className="button-bare-primary reset"
          >
            {'Reset Search'}
          </button>
        </form>
        <nav className="vertical-list-primary">
          {this.renderEntityList()}
        </nav>
        <Utility.Pagination
          pagination={this.props.pagination}
          paginationClickHandler={this.props.paginationClickHandler}
        />
      </div>
    );
  }
}
