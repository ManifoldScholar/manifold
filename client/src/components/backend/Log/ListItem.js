import React, { PureComponent } from "react";
import PropTypes from "prop-types";
import FormattedDate from "components/global/FormattedDate";
import { Link } from "react-router-dom";
import lh from "helpers/linkHandler";
import isEmpty from "lodash/isEmpty";
import humps from "humps";

export default class LogListItem extends PureComponent {
  static displayName = "Log.ListItem";

  static propTypes = {
    entity: PropTypes.object
  };

  renderChangeList(version) {
    return (
      <div className="results-desc flush">
        {`These fields were modified: `}
        <span className="specifier">
          {Object.keys(version.attributes.objectChanges)
            .map(change => humps.decamelize(change, { separator: " " }))
            .join(", ")}
        </span>
      </div>
    );
  }

  renderUserLink(version) {
    return (
      <Link to={lh.link("backendPeopleUser", version.attributes.actorId)}>
        {`${version.attributes.actorName}`}
      </Link>
    );
  }

  renderItemLink(version) {
    const urlName = `backend${version.attributes.itemType}`;
    return (
      <Link
        to={lh.link(urlName, version.attributes.itemId)}
        dangerouslySetInnerHTML={{
          __html: version.attributes.itemDisplayName
        }}
      />
    );
  }

  renderProjectLogItem(version) {
    return (
      <div className="results-body">
        <h4 className="results-header">
          <div className="subtitle">
            {this.renderUserLink(version)}
            {` updated this project `}
            <FormattedDate prefix="on" date={version.attributes.createdAt} />
          </div>
        </h4>
        {this.renderChangeList(version)}
      </div>
    );
  }

  renderLogItem(version) {
    const type = version.attributes.itemType.toLowerCase();

    return (
      <div className="results-body">
        <h4 className="results-header">
          <div className="subtitle">
            {this.renderUserLink(version)}
            {` updated the ${type}, \u201C`}
            {this.renderItemLink(version)}
            {`\u201D, `}
            <FormattedDate prefix="on" date={version.attributes.createdAt} />
          </div>
        </h4>
        {this.renderChangeList(version)}
      </div>
    );
  }

  render() {
    const version = this.props.entity;
    if (!version) return null;
    if (isEmpty(version.attributes.objectChanges)) return null;

    return (
      <li key={version.id} className="no-icon">
        {version.attributes.itemType === "Project"
          ? this.renderProjectLogItem(version)
          : this.renderLogItem(version)}
      </li>
    );
  }
}
