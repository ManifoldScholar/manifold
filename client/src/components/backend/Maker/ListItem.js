import React, { PureComponent, PropTypes } from 'react';
import { Link } from 'react-router';
import classnames from 'classnames';
import { linkHelpers as lh } from 'routes';

export default class MakerListItem extends PureComponent {

  static displayName = "Maker.ListItem";

  static propTypes = {
    entity: PropTypes.object
  };

  render() {
    const maker = this.props.entity;
    const attr = maker.attributes;
    const rowClasses = classnames({ active: this.props.active === maker.id });
    return (
      <li key={maker.id} className={rowClasses} >
        {/* Add .checked to .checkbox-primary to display checked state */}
        <div className="checkbox-primary">
          <div className="toggle-indicator">
            <i className="manicon manicon-check"></i>
          </div>
        </div>
        <Link to={lh.backendPeopleMaker(maker.id)}>
          <header>
            <figure className="avatar">
              {attr.avatarStyles.smallSquare ?
                <div
                  className="image"
                  style={ { backgroundImage: `url(${attr.avatarStyles.smallSquare})` } }
                />
                :
                <div className="no-image">
                  <i className="manicon manicon-person"></i>
                </div>
              }
            </figure>
            <div className="meta">
              <h3 className="name large">
                {attr.firstName} {attr.lastName}
              </h3>
            </div>
          </header>
          <span className="label">
          </span>
        </Link>
      </li>
    );
  }

}
