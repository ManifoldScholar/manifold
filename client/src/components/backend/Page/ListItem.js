import React, { PureComponent } from "react";
import PropTypes from "prop-types";
import { Link } from "react-router-dom";
import { Resource } from "components/frontend";
import lh from "helpers/linkHandler";

export default class PageListItem extends PureComponent {
  static displayName = "Page.ListItem";

  static propTypes = {
    entity: PropTypes.object
  };

  render() {
    const page = this.props.entity;
    const attr = page.attributes;
    return (
      <li key={page.id}>
        <Link to={lh.link("backendRecordsPage", page.id)}>
          <header>
            <figure className="asset-image">
              <div className="asset-image-placeholder">
                <Resource.Icon.Document />
              </div>
            </figure>
            <div className="meta">
              <h3 className="name large">{attr.title}</h3>
              <span className="subtitle">
                {attr.isExternalLink ? attr.externalLink : `/page/${attr.slug}`}
              </span>
            </div>
          </header>
          <span className="label">{attr.purpose.replace(/_/g, " ")}</span>
        </Link>
      </li>
    );
  }
}
