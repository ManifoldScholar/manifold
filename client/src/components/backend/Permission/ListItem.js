import React, { PureComponent } from "react";
import PropTypes from "prop-types";
import { Link } from "react-router-dom";
import lh from "helpers/linkHandler";
import classnames from "classnames";

export default class PermissionListItem extends PureComponent {
  static displayName = "Permission.ListItem";

  static propTypes = {
    entity: PropTypes.object,
    active: PropTypes.string,
    linkName: PropTypes.string.isRequired
  };

  roleName(name) {
    switch (name) {
      case "project_editor":
        return "Project Editor";
      case "project_resource_editor":
        return "Metadata Editor";
      case "project_author":
        return "Author";
      default:
        return null;
    }
  }

  renderRoles(entity) {
    const roles = entity.attributes.roleNames;
    return roles.map(role => {
      return <span key={`${entity.id}-${role}`}>{this.roleName(role)}</span>;
    });
  }

  render() {
    const entity = this.props.entity;
    if (!entity) return null;

    const resource = entity.relationships.resource;
    const user = entity.relationships.user;

    const itemClasses = classnames({
      active: this.props.active === entity.id
    });

    return (
      <li key={entity.id} className={itemClasses}>
        <Link to={lh.link(this.props.linkName, resource.id, entity.id)}>
          <header>
            <figure className="avatar">
              {user.attributes.avatarStyles.smallSquare ? (
                <div
                  className="image"
                  style={{
                    backgroundImage: `url(${
                      user.attributes.avatarStyles.smallSquare
                    })`
                  }}
                />
              ) : (
                <div className="no-image">
                  <i className="manicon manicon-person" />
                </div>
              )}
            </figure>
            <div className="meta">
              <h3 className="name large">
                {user.attributes.firstName} {user.attributes.lastName}
              </h3>
            </div>
          </header>
          <span className="label">{this.renderRoles(entity)}</span>
        </Link>
      </li>
    );
  }
}
