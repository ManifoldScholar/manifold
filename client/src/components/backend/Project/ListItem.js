import React, { PureComponent } from "react";
import PropTypes from "prop-types";
import { Link } from "react-router-dom";
import { Project as GlobalProject } from "components/global";
import { HigherOrder } from "containers/global";
import lh from "helpers/linkHandler";

export default class ProjectListItem extends PureComponent {
  static displayName = "Project.ListItem";

  static propTypes = {
    entity: PropTypes.object
  };

  projectLink(project) {
    if (
      project.attributes.abilities.update === true ||
      project.attributes.abilities.manageResources === true
    ) {
      return lh.link("backendProject", project.id);
    }
    return lh.link("frontendProject", project.id);
  }

  renderProjectMakers(makers) {
    let output = null;
    if (makers && makers.length > 0) {
      output = (
        <div className="relations-list">
          {makers.map((maker, i) => {
            let nameList = maker.attributes.fullName;
            if (i > 0) nameList = ", " + nameList;
            return nameList;
          })}
        </div>
      );
    }

    return output;
  }

  renderProjectStatusMarker(attr) {
    // Currently, this can only return a 'draft' marker
    let marker = null;

    if (attr.draft) {
      marker = <div className="block-label">{"Draft"}</div>;
    }

    return marker;
  }

  renderProjectImage(project) {
    if (project.attributes.coverStyles.smallPortrait) {
      return (
        <img
          src={project.attributes.coverStyles.smallPortrait}
          alt="Project Cover"
        />
      );
    }
    if (project.attributes.avatarStyles.smallSquare) {
      return (
        <img
          src={project.attributes.avatarStyles.smallSquare}
          alt="Project Cover"
        />
      );
    }
    return <GlobalProject.Placeholder />;
  }

  render() {
    const project = this.props.entity;
    const attr = project.attributes;
    return (
      <li key={project.id}>
        <Link to={this.projectLink(project)}>
          <header>
            <figure className="cover">
              {this.renderProjectImage(project)}
            </figure>
            <div className="meta">
              <h3 className="name">
                <span className="title-text">{attr.title}</span>
                {this.renderProjectStatusMarker(attr)}
                <span className="subtitle">{attr.subtitle}</span>
              </h3>
              {this.renderProjectMakers(project.relationships.creators)}
            </div>
          </header>
          <HigherOrder.Authorize
            successBehavior="show"
            entity={project}
            ability={["update", "manageResources"]}
          >
            <span className="label">Edit</span>
          </HigherOrder.Authorize>
          <HigherOrder.Authorize
            successBehavior="hide"
            entity={project}
            ability={["update", "manageResources"]}
          >
            <span className="label">View</span>
          </HigherOrder.Authorize>
        </Link>
      </li>
    );
  }
}
