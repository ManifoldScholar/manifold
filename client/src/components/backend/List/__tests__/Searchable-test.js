import React from "react";
import Searchable from "../Searchable";
import { shallow, render, mount } from "enzyme";
import renderer from "react-test-renderer";
import ListItem from "components/backend/Resource/ListItem";
import build from "test/fixtures/build";
import { wrapWithRouter, renderWithRouter } from "test/helpers/routing";

describe("Backend.List.Searchable component", () => {
  const entities = [build.entity.resource("1"), build.entity.resource("2")];
  const pagination = {
    perPage: 1,
    currentPage: 1,
    totalPages: 10,
    totalCount: 2,
    nextPage: 2,
    prevPage: 0
  };
  const pageChangeMock = jest.fn();
  const filterChangeMock = jest.fn();
  const fakeClick = {
    stopPropagation: () => undefined,
    preventDefault: () => undefined
  };
  const root = wrapWithRouter(
    <Searchable
      entities={entities}
      singularUnit="resource"
      pluralUnit="resources"
      pagination={pagination}
      paginationClickHandler={() => pageChangeMock}
      entityComponent={ListItem}
      filterChangeHandler={filterChangeMock}
      filterOptions={{
        type: ["TWEET", "PROJECT_CREATED"]
      }}
    />
  );

  function buildComponent(props) {
    return (
      <div>
        Can we build {props.entity.attributes.title}?
        <p>Yes, we can.</p>
      </div>
    );
  }

  it("renders correctly", () => {
    const component = renderer.create(root);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly when passed a component builder function", () => {
    const component = renderer.create(
      wrapWithRouter(
        <Searchable
          entities={entities}
          singularUnit="resource"
          pluralUnit="resources"
          pagination={pagination}
          paginationClickHandler={() => pageChangeMock}
          entityComponent={buildComponent}
          filterChangeHandler={filterChangeMock}
          filterOptions={{
            type: ["TWEET", "PROJECT_CREATED"]
          }}
        />
      )
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render the correct number of ListItems", () => {
    const wrapper = mount(root);
    expect(wrapper.find(ListItem)).toHaveLength(2);
  });

  it("should show filter options when toggle is clicked", () => {
    const wrapper = mount(root);
    expect(wrapper.find('[data-id="filter"]')).toHaveLength(0);
    wrapper.find('[data-id="filter-toggle"]').simulate("click", fakeClick);
    expect(wrapper.find('[data-id="filter"]')).toHaveLength(1);
  });

  it("should trigger filterChangeMock callback when filter is changed", () => {
    const wrapper = mount(root);
    const filterEvent = {
      preventDefault: () => true,
      target: {
        value: "TWEET"
      }
    };
    wrapper.find('[data-id="filter-toggle"]').simulate("click", fakeClick);
    wrapper.find('[data-id="filter"]').simulate("change", filterEvent);
    expect(filterChangeMock).toHaveBeenCalled();
  });

  it("should trigger paginationClickHandler callback when page change is clicked", () => {
    const wrapper = mount(root);
    pageChangeMock.mockClear();
    wrapper.find('[data-id="page-next"]').simulate("click", fakeClick);
    expect(pageChangeMock).toHaveBeenCalled();
  });
});
