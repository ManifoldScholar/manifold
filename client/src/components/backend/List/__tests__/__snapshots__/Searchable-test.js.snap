// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Backend.List.Searchable component renders correctly 1`] = `
<div>
  <form
    className="form-search-filter"
    onSubmit={[Function]}
  >
    <div
      className="search"
    >
      <button>
        <i
          className="manicon manicon-magnify"
        />
        <span
          className="screen-reader-text"
        >
          Click to search
        </span>
      </button>
      <input
        onChange={[Function]}
        placeholder="Search..."
        type="text"
        value=""
      />
    </div>
    <div
      className="form-list-filter"
    >
      <div
        className="select-group"
      />
    </div>
    <button
      className="button-bare-primary"
      data-id="filter-toggle"
      onClick={[Function]}
    >
      Show Search Options
    </button>
    <button
      className="button-bare-primary reset"
      onClick={[Function]}
    >
      Reset Search
    </button>
  </form>
  <nav
    className="vertical-list-primary"
  >
    <div>
      <p
        className="list-total"
      >
        Showing 
        <span>
          1-1
        </span>
         of 
        <span>
          2
        </span>
         resources: 
      </p>
      <div
        className="buttons-icon-horizontal"
      />
      <section>
        <ul>
          <li>
            <a
              href="/backend/resource/1"
              onClick={[Function]}
            >
              <header>
                <figure
                  className="cover"
                >
                  <div
                    className="icon-thumbnail-primary icon-only"
                    style={
                      Object {
                        "backgroundImage": null,
                      }
                    }
                  >
                    <div
                      className="wrapper"
                    >
                      <figure
                        className="icon-thumbnail-type"
                      >
                        <i
                          className="icon-thumbnail-icon image"
                        >
                          <svg
                            height="72px"
                            version="1.1"
                            viewBox="0 0 72 72"
                            width="72px"
                          >
                            <path
                              d="M69.82031,9.86816H2.17969c-0.55225,0-1,0.44775-1,1v50.26367c0,0.55273,0.44775,1,1,1h67.64062c0.55273,0,1-0.44727,1-1 V10.86816C70.82031,10.31592,70.37305,9.86816,69.82031,9.86816z M68.82031,11.86816v42.29572l-7.68848-10.3631 c-0.9248-1.24707-2.40918-2.03027-4.07227-2.15137c-1.7002-0.12012-3.33691,0.46094-4.47656,1.60449l-6.87012,6.89551 L29.42676,30.76709c-1.00781-1.19873-2.54541-1.90332-4.21924-1.93262c-1.71143-0.05127-3.2583,0.62695-4.31787,1.80127 L3.17969,50.271V11.86816H68.82031z M3.17969,60.13184v-6.87415L22.375,31.97559c0.67188-0.74561,1.7002-1.1582,2.79736-1.1416 c1.09521,0.01953,2.08789,0.46387,2.72314,1.22021l16.98926,20.21826c0.18066,0.21484,0.44336,0.34375,0.72363,0.35547 c0.27637,0.01953,0.55273-0.09473,0.75-0.29297l7.64062-7.66895c0.72949-0.73145,1.78711-1.10449,2.91602-1.02148 c1.08203,0.07812,2.0332,0.56934,2.61133,1.34766l9.29395,12.5271v2.61255H3.17969z"
                            />
                            <path
                              d="M52.21191,34.1792c4.0166,0,7.28418-3.26758,7.28418-7.28418s-3.26758-7.28418-7.28418-7.28418 s-7.28418,3.26758-7.28418,7.28418S48.19531,34.1792,52.21191,34.1792z M52.21191,21.61084 c2.91406,0,5.28418,2.37061,5.28418,5.28418s-2.37012,5.28418-5.28418,5.28418s-5.28418-2.37061-5.28418-5.28418 S49.29785,21.61084,52.21191,21.61084z"
                            />
                          </svg>
                        </i>
                      </figure>
                    </div>
                  </div>
                </figure>
                <div
                  className="meta"
                >
                  <h3
                    className="name"
                  >
                    <span
                      dangerouslySetInnerHTML={
                        Object {
                          "__html": "Image",
                        }
                      }
                    />
                    <span
                      className="subtitle"
                    >
                      <span>
                        April 24, 2017
                      </span>
                    </span>
                  </h3>
                </div>
              </header>
            </a>
          </li>
          <li>
            <a
              href="/backend/resource/2"
              onClick={[Function]}
            >
              <header>
                <figure
                  className="cover"
                >
                  <div
                    className="icon-thumbnail-primary icon-only"
                    style={
                      Object {
                        "backgroundImage": null,
                      }
                    }
                  >
                    <div
                      className="wrapper"
                    >
                      <figure
                        className="icon-thumbnail-type"
                      >
                        <i
                          className="icon-thumbnail-icon image"
                        >
                          <svg
                            height="72px"
                            version="1.1"
                            viewBox="0 0 72 72"
                            width="72px"
                          >
                            <path
                              d="M69.82031,9.86816H2.17969c-0.55225,0-1,0.44775-1,1v50.26367c0,0.55273,0.44775,1,1,1h67.64062c0.55273,0,1-0.44727,1-1 V10.86816C70.82031,10.31592,70.37305,9.86816,69.82031,9.86816z M68.82031,11.86816v42.29572l-7.68848-10.3631 c-0.9248-1.24707-2.40918-2.03027-4.07227-2.15137c-1.7002-0.12012-3.33691,0.46094-4.47656,1.60449l-6.87012,6.89551 L29.42676,30.76709c-1.00781-1.19873-2.54541-1.90332-4.21924-1.93262c-1.71143-0.05127-3.2583,0.62695-4.31787,1.80127 L3.17969,50.271V11.86816H68.82031z M3.17969,60.13184v-6.87415L22.375,31.97559c0.67188-0.74561,1.7002-1.1582,2.79736-1.1416 c1.09521,0.01953,2.08789,0.46387,2.72314,1.22021l16.98926,20.21826c0.18066,0.21484,0.44336,0.34375,0.72363,0.35547 c0.27637,0.01953,0.55273-0.09473,0.75-0.29297l7.64062-7.66895c0.72949-0.73145,1.78711-1.10449,2.91602-1.02148 c1.08203,0.07812,2.0332,0.56934,2.61133,1.34766l9.29395,12.5271v2.61255H3.17969z"
                            />
                            <path
                              d="M52.21191,34.1792c4.0166,0,7.28418-3.26758,7.28418-7.28418s-3.26758-7.28418-7.28418-7.28418 s-7.28418,3.26758-7.28418,7.28418S48.19531,34.1792,52.21191,34.1792z M52.21191,21.61084 c2.91406,0,5.28418,2.37061,5.28418,5.28418s-2.37012,5.28418-5.28418,5.28418s-5.28418-2.37061-5.28418-5.28418 S49.29785,21.61084,52.21191,21.61084z"
                            />
                          </svg>
                        </i>
                      </figure>
                    </div>
                  </div>
                </figure>
                <div
                  className="meta"
                >
                  <h3
                    className="name"
                  >
                    <span
                      dangerouslySetInnerHTML={
                        Object {
                          "__html": "Image",
                        }
                      }
                    />
                    <span
                      className="subtitle"
                    >
                      <span>
                        April 24, 2017
                      </span>
                    </span>
                  </h3>
                </div>
              </header>
            </a>
          </li>
        </ul>
      </section>
    </div>
  </nav>
  <nav
    className="list-pagination-primary"
  >
    <ul>
      <li
        className="pagination-previous disabled"
      >
        <a
          data-id="page-prev"
          href="#pagination-target"
          onClick={[MockFunction]}
        >
          <i
            className="manicon manicon-arrow-long-left"
          />
          <span>
            Prev
          </span>
        </a>
      </li>
      <li
        className="active ordinal"
      >
        <a
          href="#pagination-target"
          onClick={[MockFunction]}
        >
          1
        </a>
      </li>
      <li
        className="ordinal"
      >
        <a
          href="#pagination-target"
          onClick={[MockFunction]}
        >
          2
        </a>
      </li>
      <li
        className="ordinal"
      >
        <a
          href="#pagination-target"
          onClick={[MockFunction]}
        >
          3
        </a>
      </li>
      <li
        className="ordinal"
      >
        <a
          href="#pagination-target"
          onClick={[MockFunction]}
        >
          4
        </a>
      </li>
      <li
        className="pagination-next "
      >
        <a
          data-id="page-next"
          href="#pagination-target"
          onClick={[MockFunction]}
        >
          <span>
            Next
          </span>
          <i
            className="manicon manicon-arrow-long-right"
          />
        </a>
      </li>
    </ul>
  </nav>
</div>
`;

exports[`Backend.List.Searchable component renders correctly when passed a component builder function 1`] = `
<div>
  <form
    className="form-search-filter"
    onSubmit={[Function]}
  >
    <div
      className="search"
    >
      <button>
        <i
          className="manicon manicon-magnify"
        />
        <span
          className="screen-reader-text"
        >
          Click to search
        </span>
      </button>
      <input
        onChange={[Function]}
        placeholder="Search..."
        type="text"
        value=""
      />
    </div>
    <div
      className="form-list-filter"
    >
      <div
        className="select-group"
      />
    </div>
    <button
      className="button-bare-primary"
      data-id="filter-toggle"
      onClick={[Function]}
    >
      Show Search Options
    </button>
    <button
      className="button-bare-primary reset"
      onClick={[Function]}
    >
      Reset Search
    </button>
  </form>
  <nav
    className="vertical-list-primary"
  >
    <div>
      <p
        className="list-total"
      >
        Showing 
        <span>
          1-1
        </span>
         of 
        <span>
          2
        </span>
         resources: 
      </p>
      <div
        className="buttons-icon-horizontal"
      />
      <section>
        <ul>
          <div>
            Can we build 
            Image
            ?
            <p>
              Yes, we can.
            </p>
          </div>
          <div>
            Can we build 
            Image
            ?
            <p>
              Yes, we can.
            </p>
          </div>
        </ul>
      </section>
    </div>
  </nav>
  <nav
    className="list-pagination-primary"
  >
    <ul>
      <li
        className="pagination-previous disabled"
      >
        <a
          data-id="page-prev"
          href="#pagination-target"
          onClick={[MockFunction]}
        >
          <i
            className="manicon manicon-arrow-long-left"
          />
          <span>
            Prev
          </span>
        </a>
      </li>
      <li
        className="active ordinal"
      >
        <a
          href="#pagination-target"
          onClick={[MockFunction]}
        >
          1
        </a>
      </li>
      <li
        className="ordinal"
      >
        <a
          href="#pagination-target"
          onClick={[MockFunction]}
        >
          2
        </a>
      </li>
      <li
        className="ordinal"
      >
        <a
          href="#pagination-target"
          onClick={[MockFunction]}
        >
          3
        </a>
      </li>
      <li
        className="ordinal"
      >
        <a
          href="#pagination-target"
          onClick={[MockFunction]}
        >
          4
        </a>
      </li>
      <li
        className="pagination-next "
      >
        <a
          data-id="page-next"
          href="#pagination-target"
          onClick={[MockFunction]}
        >
          <span>
            Next
          </span>
          <i
            className="manicon manicon-arrow-long-right"
          />
        </a>
      </li>
    </ul>
  </nav>
</div>
`;
